src/tools/cephfs/JournalTool.cc

```diff
@@ -250,17 +257,36 @@ int JournalTool::main_journal(std::vector<const char*> &argv)
     }
   } else if (command == "reset") {
     bool force = false;
-    if (argv.size() == 2) {
+    if (argv.size() == 1) {
+        std::cerr << "warning: this operation resets the journal!!!\n"
+                  << "         if you want to continue, please add --yes-i-really-really-mean-it!!!"
+                  << std::endl;
+        return -EINVAL;
+    } else if (argv.size() == 2) {
+      if (std::string(argv[1]) != "--yes-i-really-really-mean-it") {
+        std::cerr << "warning: this operation resets the journal!!!\n"
+                  << "         if you want to continue, please add --yes-i-really-really-mean-it!!!"
+                  << std::endl;
+        return -EINVAL;
+      }
+    } else if (argv.size() == 3) {
       if (std::string(argv[1]) == "--force") {
         force = true;
       } else {
         std::cerr << "Unknown argument " << argv[1] << std::endl;
         return -EINVAL;
       }
-    } else if (argv.size() > 2) {
+      if (std::string(argv[2]) == "" || std::string(argv[2]) != "--yes-i-really-really-mean-it") {
```

Comment:
What does checking for empty string do here? Do we need it for `argv.size() == 2`?

Reply:
Yes, that's unnecessary check. Fixed it.